common/src/main/java/com/microsoft/identity/common/internal/cache/AccountCredentialCacheKeyValueDelegate.java

```diff
@@ -39,32 +46,31 @@ public AccountCredentialCacheKeyValueDelegate() {
         mGson = new Gson();
     }
 
+    static class CacheKeyReplacements {
+        static final String UNIQUE_USER_ID = "<unique_user_id>";
+        static final String ENVIRONMENT = "<environment>";
+        static final String REALM = "<realm>";
+        static final String CREDENTIAL_TYPE = "<credential_type>";
+        static final String CLIENT_ID = "<client_id>";
+        static final String TARGET = "<target>";
+    }
 
-    protected final String collapseKeyComponents(final List<String> keyComponents) {
-        String cacheKey = "";
-
-        for (String keyComponent : keyComponents) {
-            if (!StringExtensions.isNullOrBlank(keyComponent)) {
-                keyComponent = keyComponent.toLowerCase(Locale.US);
-                cacheKey += keyComponent + CACHE_VALUE_SEPARATOR;
-            }
-        }
-
-        if (cacheKey.endsWith(CACHE_VALUE_SEPARATOR)) {
-            cacheKey = cacheKey.substring(0, cacheKey.length() - 1);
-        }
-
-        return cacheKey;
+    private static String sanitizeNull(final String input) {
+        return null == input ? "" : input.toLowerCase(Locale.US);
     }
 
     @Override
     public String generateCacheKey(Account account) {
-        final List<String> keyComponents = new ArrayList<>();
-        keyComponents.add(account.getUniqueId());
-        keyComponents.add(account.getEnvironment());
-        keyComponents.add(account.getRealm());
+        String cacheKey = UNIQUE_USER_ID
+                + CACHE_VALUE_SEPARATOR
+                + ENVIRONMENT
+                + CACHE_VALUE_SEPARATOR
+                + REALM;
+        cacheKey = cacheKey.replace(UNIQUE_USER_ID, sanitizeNull(account.getUniqueUserId()));
```

Comment:
might be a silly question, but wouldn't replacing null with empty string break backwards compat?

Reply:
So this particular logic speaks exclusively to the new MSAL cache. As there's no old-MSAL cache to common cache migration story, this isn't breaking. Users who take the new common-cache-enabled MSAL are expected to sign-in again when they get the new version